src/Diagnostics/FxCop/Core/Usage/SerializationRulesDiagnosticAnalyzer.cs

```diff
@@ -186,7 +190,9 @@ public void AnalyzeSymbol(SymbolAnalysisContext context)
 
             private bool IsSerializable(ITypeSymbol namedTypeSymbol)
             {
-                return namedTypeSymbol.GetAttributes().Any(a => a.AttributeClass == _serializableAttributeTypeSymbol);
+                return namedTypeSymbol.IsValueType ||
```

Comment:
With this you are suppressing this check for all value types - so all structs as well. If you want to suppress just primitive types you'll have to check for that explicitly something like - http://source.roslyn.io/#System.Runtime.Analyzers/Reliability/DoNotLockOnObjectsWithWeakIdentity.cs,92432dae4f408c01